gen/definitions/device_vtep_policy.yaml

```diff
@@ -0,0 +1,110 @@
+---
+name: Device VTEP Policy
+rest_endpoint: /api/fmc_config/v1/domain/{DOMAIN_UUID}/devices/devicerecords/%v/vteppolicies
+res_description: >-
+  This resource can manage the Device VTEP Policy. Practicioners should ensure only one resource
+  `fmc_device_vtep_policy` exists for a single `fmc_device`, because the FMC API responds with the same singe UUID for
+  every request to create a new VTEP Policy on the same Device.
+  Thus multiple resources per Device would unexpectedly overwrite the same set of settings.
+doc_category: Devices
+test_tags: [TF_VAR_device_id]
+attributes:
+  - tf_name: device_id
+    type: String
+    reference: true
+    requires_replace: true
+    description: >-
+      UUID of the parent device (fmc_device.example.id).
+    example: 76d24097-41c4-4558-a4d0-a8c07ac08470
+    test_value: fmc_device_physical_interface.test.device_id
+  - model_name: nveEnable
+    tf_name: nve_enabled
+    description: Indicates whether to enable NVE on the `device_id`. Can only be false if `vteps` are empty.
+    type: Bool
+    default_value: "true"
+    exclude_example: true
+    minimum_test_value: "false"
+    test_value: "true"
+  - model_name: vtepEntries
+    tf_name: vteps
+    type: List
+    description: List that can either be empty or contain one VTEP object.
```

Comment:
I think we don't want empty lists (instead of null) because "read" (fromBodyPartial) will not create empty lists and "plan" will therefore show a diff.

Reply:
Could you be more specific about the scenario? The fromBodyPartial keeps the old list when it's either null or empty. I've did a config with `vteps = []` and with `vteps = null` and tried them both:

```
$ terraform apply -refresh=true
...
Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
$ terraform apply -refresh=true
fmc_device_vtep_policy.example: Refreshing state... [id=0050568A-FBD5-0ed3-0000-004295408043]

No changes. Your infrastructure still matches the configuration.
```

Reply:
Right, I have confused this with a different implementation.